src/views/studio/studio-info.jsx

```diff
@@ -1,35 +1,54 @@
-import React, {useState, useEffect} from 'react';
+import React, {useEffect} from 'react';
+import PropTypes from 'prop-types';
 import {useParams} from 'react-router-dom';
-import {infoFetcher} from './lib/fetchers';
+import {connect} from 'react-redux';
 import Debug from './debug.jsx';
+import {getInfo, getRoles} from '../../redux/studio';
 
-const StudioInfo = () => {
+const StudioInfo = ({username, studio, token, onLoadInfo, onLoadRoles}) => {
     const {studioId} = useParams();
-    const [state, setState] = useState({loading: false, error: null, data: null});
-    // Since this data is in a component that is always visible, it doesn't necessarily
-    // need to be kept in redux. One alternative is to use the infinite-list redux
-    // module and just treat the studio info as the first and only item in the list.
-    useEffect(() => {
-        if (!studioId) return;
-        infoFetcher(studioId)
-            .then(data => setState({loading: false, error: null, data}))
-            .catch(error => setState({loading: false, error, data: null}));
+    
+    useEffect(() => { // Load studio info after first render
+        if (studioId) onLoadInfo(studioId);
     }, [studioId]);
 
+    useEffect(() => { // Load roles info once the username is available
+        if (studioId && username && token) onLoadRoles(studioId, username, token);
+    }, [studioId, username, token]);
```

Comment:
Just curious — would it continue working fine if you only gave it `[studioId, username]`? And this causes it to rerender when you log in/out?

Reply:
> would it continue working fine if you only gave it [studioId, username]?
Since username/token are updated together, yeah I think it would, but I thought that knowledge isn't really represented in the code so I should make the dependency explicit. 

> this causes it to rerender when you log in/out?
I think right now our logout redirects to the homepage, but if it was done without refresh it would cause onLoadRoles to be called again.